File: app/src/main/java/com/quranapp/android/utils/tafsir/TafsirJSInterface.java
package com.quranapp.android.utils.tafsir;

import android.content.ComponentName;
import android.content.Intent;
import android.webkit.JavascriptInterface;

import com.quranapp.android.activities.ActivityReader;
import com.quranapp.android.activities.ActivityTafsir2;
import com.quranapp.android.utils.reader.factory.ReaderFactory;
import com.quranapp.android.utils.univ.Codes;
import com.quranapp.android.utils.univ.Keys;

public class TafsirJSInterface {
    private final ActivityTafsir2 mActivityTafsir;

    public TafsirJSInterface(ActivityTafsir2 activityTafsir) {
        mActivityTafsir = activityTafsir;
    }

    @JavascriptInterface
    public void goToVerse() {
        Intent intent = ReaderFactory.prepareSingleVerseIntent(mActivityTafsir.mChapterNo, mActivityTafsir.mVerseNo);

        ComponentName callingActivity = mActivityTafsir.getCallingActivity();
        if (callingActivity != null && ActivityReader.class.getName().equals(callingActivity.getClassName())) {
            mActivityTafsir.setResult(Codes.OPEN_REFERENCE_RESULT_CODE, intent);
            mActivityTafsir.finish();
        } else {
            intent = intent.setClass(mActivityTafsir, ActivityReader.class);
            mActivityTafsir.startActivity(intent);
        }
    }

    @JavascriptInterface
    public void previousTafsir() {
        if (mActivityTafsir.mVerseNo == 1) {
            return;
        }
        Intent intent = mActivityTafsir.getIntent();
        intent.putExtra(Keys.READER_KEY_VERSE_NO, mActivityTafsir.mVerseNo - 1);
        intent.setAction(null);
        mActivityTafsir.startActivity(intent);
    }

    @JavascriptInterface
    public void nextTafsir() {
        if (mActivityTafsir.mVerseNo == mActivityTafsir.mQuranMeta.getChapterVerseCount(mActivityTafsir.mChapterNo)) {
            return;
        }

        Intent intent = mActivityTafsir.getIntent();
        intent.putExtra(Keys.READER_KEY_VERSE_NO, mActivityTafsir.mVerseNo + 1);
        intent.setAction(null);
        mActivityTafsir.startActivity(intent);
    }
}

File: app/src/main/java/com/quranapp/android/utils/tafsir/TafsirJsInterface.kt
package com.quranapp.android.utils.tafsir

import android.webkit.JavascriptInterface
import com.quranapp.android.activities.ActivityReader
import com.quranapp.android.activities.ActivityTafsir
import com.quranapp.android.utils.reader.factory.ReaderFactory.prepareSingleVerseIntent
import com.quranapp.android.utils.univ.Codes
import com.quranapp.android.utils.univ.Keys

class TafsirJsInterface(private val activity: ActivityTafsir) {
    @JavascriptInterface
    fun goToVerse() {
        var intent = prepareSingleVerseIntent(activity.chapterNo, activity.verseNo)

        val callingActivity = activity.callingActivity
        if (callingActivity != null && ActivityReader::class.java.name == callingActivity.className) {
            activity.setResult(Codes.OPEN_REFERENCE_RESULT_CODE, intent)
            activity.finish()
        } else {
            intent = intent.setClass(activity, ActivityReader::class.java)
            activity.startActivity(intent)
        }
    }


    @JavascriptInterface
    fun previousTafsir() {
        if (activity.verseNo == 1) {
            return
        }

        val intent = activity.intent.also {
            it.putExtra(Keys.READER_KEY_VERSE_NO, activity.verseNo - 1)
            it.action = null
        }

        activity.startActivity(intent)
    }

    @JavascriptInterface
    fun nextTafsir() {
        if (activity.verseNo == activity.mQuranMetaRef.get()!!.getChapterVerseCount(activity.chapterNo)) {
            return
        }

        val intent = activity.intent.also {
            it.putExtra(Keys.READER_KEY_VERSE_NO, activity.verseNo + 1)
            it.action = null
        }

        activity.startActivity(intent)
    }

    @JavascriptInterface
    fun goToTop() {
        activity.scrollToTop()
    }
}